addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200GatewayHandler.java

```diff
@@ -0,0 +1,710 @@
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.km200.KM200ServiceType;
+import org.openhab.binding.km200.KM200ThingType;
+import org.openhab.binding.km200.internal.KM200Comm;
+import org.openhab.binding.km200.internal.KM200CommObject;
+import org.openhab.binding.km200.internal.KM200Device;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.net.InetAddresses;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link KM200GatewayHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Markus Eckhardt - Initial contribution
+ */
+public class KM200GatewayHandler extends BaseBridgeHandler {
+
+    private final static Logger logger = LoggerFactory.getLogger(KM200GatewayHandler.class);
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Collections.singleton(THING_TYPE_KMDEVICE);
+
+    private Map<Channel, byte[]> sendMap = Collections.synchronizedMap(new LinkedHashMap<Channel, byte[]>());
+
+    private GetKM200Runnable runnable;
+    private ScheduledFuture<?> pollingJob;
+    private final KM200Device device;
+    private final KM200Comm<KM200Device> comm;
+    private SendKM200Thread sThread;
+    private static BigDecimal readDelay;
+
+    public KM200GatewayHandler(Bridge bridge) {
+        super(bridge);
+        readDelay = BigDecimal.valueOf(100);
+        thing.setStatusInfo(
+                new ThingStatusInfo(ThingStatus.UNINITIALIZED, ThingStatusDetail.CONFIGURATION_PENDING, ""));
+
+        device = new KM200Device();
+        comm = new KM200Comm<KM200Device>(getDevice());
+    }
+
+    private List<KM200GatewayStatusListener> listeners = new CopyOnWriteArrayList<KM200GatewayStatusListener>();
```

Comment:
Please put all fields on top, then its more clear what the internal state of this object contains.